index.ts

```diff
@@ -54,6 +54,69 @@ const listOrganizationRepositoriesTool: Tool = {
   },
 };
 
+const searchRepositoryIssuesTool: Tool = {
+  name: 'codacy_list_repository_issues',
+  description: 'List issues in a repository with pagination',
+  inputSchema: {
+    type: 'object',
+    properties: {
+      provider: {
+        type: 'string',
+        description:
+          "Repository's git provider: GitHub (gh), GitLab (gl) or BitBucket (bb). Accepted values: gh, gl, bb.",
+      },
+      organization: {
+        type: 'string',
+        description: 'Organization name, owner of the repository',
+      },
+      repository: {
+        type: 'string',
+        description: 'Repository name',
+      },
+      cursor: {
+        type: 'string',
+        description: 'Pagination cursor for next page of results',
+      },
+      limit: {
+        type: 'number',
+        description: 'Maximum number of results to return (default 100, max 100)',
+        default: 100,
+      },
+      body: {
+        type: 'object',
+        description: 'Search parameters to filter the list of issues in a repository',
+        properties: {
+          branchName: {
+            type: 'string',
+            description:
+              'Branch name, by default the main branch defined on the Codacy repository settings is used',
+          },
+          patternIds: {
+            type: 'array',
+            description: 'Set of code pattern identifiers',
+          },
```

Comment:
I wonder how the AI can use this field 🤔

Reply:
right... it would probably need to fetch the patterns first, find the ids, and then using this